src/farcasterd/runtime.rs

```diff
@@ -2042,32 +1140,20 @@ fn syncers_up(
         args.append(&mut syncer_servers_args(config, blockchain, network)?);
         info!("launching syncer with: {:?}", args);
         launch("syncerd", args)?;
-        syncers.insert(
-            k,
-            Syncer {
-                clients: none!(),
-                service_id: None,
-            },
-        );
-        spawning_services.insert(s, source);
+        spawning_services.insert(syncer_service.clone());
     }
-    if let Some(xs) = syncers.get_mut(&k) {
-        xs.clients.insert(ServiceId::Swap(swap_id));
+    if registered_services.contains(&syncer_service) {
+        Ok(Some(syncer_service))
+    } else {
+        Ok(None)
     }
-    Ok(())
 }
 
 #[allow(clippy::too_many_arguments)]
-fn launch_swapd(
-    runtime: &mut Runtime,
-    peerd: ServiceId,
-    report_to: Option<ServiceId>,
+pub fn launch_swapd(
     local_trade_role: TradeRole,
     public_offer: PublicOffer,
-    local_params: Params,
     swap_id: SwapId,
-    remote_commit: Option<Commit>,
-    funding_address: Option<bitcoin::Address>,
 ) -> Result<String, Error> {
```

Comment:
great that these are now minimized (of course, still pending a potential merge of swap id & offer uuids)